nodeJS/express/introduction_to_express.md

```diff
@@ -0,0 +1,98 @@
+
+Whenever you navigate to any web URL this way, this is essentially what you are doing. Navigating to `https://theodinproject.com/paths` via the address bar is just telling the browser to send a `GET` request to the `/paths` route at `https://theodinproject.com`, then display what it receives in response.
+
+Once you navigate to `http://localhost:3000/`, you should see `Hello, world!` appear in the window. Magic, right?
+
+When our server receives our `GET` request, Express stores the request in a [request object](https://expressjs.com/en/4x/api.html#req). This request gets passed through a chain of functions we call `middleware functions` until eventually, a middleware function tells Express to respond to the request.
+
+In our example, the request comes through as a `GET` request to the `/` route. This matches the route handler we have in our `app.js` file.
+
+```javascript
+app.get("/", (req, res) => res.send("Hello, world!"));
+```
+
+We will discuss routes and route handlers in more detail in a later lesson, but to summarise the above line, it tells Express "if a `GET` request comes through to the `/` route, run the request through the following chain of middleware functions". Here, we only have a single function.
+
+If we had defined multiple route handlers, Express would pass the request through the first one to match the requested route. Order matters!
+
+Express takes the callback function we gave it and passes the request object into the first parameter (commonly named `req` by convention), and a [response object](https://expressjs.com/en/4x/api.html#res) into the second parameter (commonly named `res`). Our callback tells the response object to respond to the request by `.send`ing the string `"Hello, world!"`.
+
+There is no more code to run and the function returns. Since Express has been told to respond to the request, it ends the request here. Meanwhile, the browser receives our server's response and displays it on screen, which is our `"Hello, world!"` string. We could send nearly anything in response. We could even [tell Express to send a file](https://expressjs.com/en/api.html#res.sendFile).
+
+### Nodemon
+
+After you run your server with `node app.js`, if you make any changes to any JavaScript and JSON files in the project directory, these changes will not be reflected in the server unless you manually interrupt and rerun `node app.js`. Similarly to how Webpack and Vite's dev servers would restart and rebuild your site when you make a change to a file, you can use [Nodemon](https://www.npmjs.com/package//nodemon) to run your `app.js` file.
```

Comment:
```suggestion
When you run your server with `node app.js`, any changes to JavaScript and JSON files in your project directory won't be reflected automatically unless you manually interrupt and rerun `node app.js`. To avoid this manual process, you can use [Nodemon](https://www.npmjs.com/package/nodemon) to run your "app.js" file. Nodemon automatically restarts and rebuilds your site when you make changes to files, similar to Webpack and Vite's dev servers.
```